blog/views.py

```diff
@@ -1,40 +1,41 @@
-from django.shortcuts import render, get_object_or_404
+from django.shortcuts import render, redirect, get_object_or_404
 from django.utils import timezone
 from .models import Post 
 from .forms import PostForm
 
+
 # Create your views here.
 def post_list(request):
-	posts = Post.objects.filter(published_date__lte=timezone.now()).order_by('published_date')
-	return render(request, 'blog/post_list.html', {'posts' : posts})
+        posts = Post.objects.filter(published_date__lte=timezone.now()).order_by('published_date')
+        return render(request, 'blog/post_list.html', {'posts' : posts})
 
 def post_detail(request, pk):
-	post = get_object_or_404(Post, pk=pk)
-	return render(request, 'blog/post_detail.html', {'post': post})
+        post = get_object_or_404(Post, pk=pk)
+        return render(request, 'blog/post_detail.html', {'post': post})
 
 def post_new(request):
-	if request.method == "POST":
-		form = PostForm(REQUEST.POST)
-		if form.is_valid():
-			post= form.save(commit=False)
-			post.author = request.user
-			post.published_date = timezone.now()
-			post.save
-			return redirect('blog.views.post_detail', pk=post.pk)
-	else:
-		form = PostForm()
-	return render(request, 'blog/post_edit.html', {'form': form})
+        if request.method == "POST":
+                form = PostForm(request.POST)
```

Comment:
In line 16 of this views file, you pass a parameter named `request`. In this line you were calling `REQUEST`, which does not exist, since Python is case sensitive. This fixes the NameError you hit. :)